### PR TITLE
Fail on using the context manager

### PR DESCRIPTION
**Pull request by anijain2305**
I have removed the context manager usage in PyTorch CI - https://github.com/pytorch/pytorch/pull/85124

Internally, there are 3 files where context manager is used. And I am talking to migrate to use annotation.

With this, we can remove the context manager.

### CODE REVIEW

**Review by voznesenskym — APPROVED** [review]

**Comment by anijain2305** [comment]
@jansel Wondering if you are ok with this. We should be done with internal migration soon.

File: torchdynamo/eval_frame.py [hunk 1]
```diff
@@ -104,15 +104,16 @@ def __init__(
...
-        log.warning(
-            (
-                "Using TorchDynamo with a context manager will be deprecated soon."
-                "Please read https://github.com/pytorch/torchdynamo#usage-example "
-                "to use TorchDynamo using an annotation."
-            )
-        )
         if config.raise_on_ctx_manager_usage:
             raise RuntimeError("TorchDynamo is used with a context manager")
+        else:
```
Comment (jansel) [inline torchdynamo/eval_frame.py:109]: Should we just remove the warning now that raising is the default behavior?  If someone sets the config they should know what they are doing and the warning will just be annoying

File: torchdynamo/eval_frame.py [hunk 2]
```diff
@@ -104,15 +104,16 @@ def __init__(
...
     def __enter__(self):
-        log.warning(
-            (
-                "Using TorchDynamo with a context manager will be deprecated soon."
-                "Please read https://github.com/pytorch/torchdynamo#usage-example "
-                "to use TorchDynamo using an annotation."
-            )
-        )
         if config.raise_on_ctx_manager_usage:
             raise RuntimeError("TorchDynamo is used with a context manager")
```
Comment (jansel) [inline torchdynamo/eval_frame.py:108]: I'd suggest some more context in this error string that perhaps mentions the correct usage.

**Review by jansel — COMMENTED** [review]
I am fine with this, a few comments.

File: torchdynamo/eval_frame.py (hunk 2 above)
Comment (anijain2305) [reply torchdynamo/eval_frame.py:108]: Will do.

File: torchdynamo/eval_frame.py (hunk 1 above)
Comment (anijain2305) [reply torchdynamo/eval_frame.py:109]: Yeah, thats correct. Will remove it.

**Pull request merged**